src/cleanup.ml

```diff
@@ -175,6 +207,5 @@ end
 
 module Persistent = struct
   let cleanup () =
-    local_login () >>= fun (rpc, session_id) ->
-    VBD.Persistent.cleanup rpc session_id
+    Local_xapi_session.with_session VBD.Persistent.cleanup
```

Comment:
I could also change the code to only log in when the list of VBDs/block devices to clean up is non-empty, but thought it's better to keep cleanup code clean and simple, because this has to work.